modules/sql-engine/src/integrationTest/sql/types/blob/test_blob_cast.test

```diff
@@ -63,3 +63,5 @@ SELECT 1::bigint::VARBINARY
 statement error
 SELECT 1::decimal::VARBINARY
 
+statement error
+select CAST(x'Aa' AS integer);
```

Comment:
BTW, this valid valid in 2023 SQL standard.
ISO/IEC 9075-2:2023 as the optional feature T661, “Non-decimal integer literals”